lib/ex_aws/sts/parsers.ex

```diff
@@ -2,6 +2,30 @@ if Code.ensure_loaded?(SweetXml) do
   defmodule ExAws.STS.Parsers do
     import SweetXml, only: [sigil_x: 2]
 
+    def parse({:ok, %{body: xml}=resp}, :assume_role) do
+      parsed_body = xml
+      |> SweetXml.xpath(~x"//AssumeRoleResponse",
+                        access_key_id: ~x"./AssumeRoleResult/Credentials/AccessKeyId/text()"s,
+                        secret_access_key: ~x"./AssumeRoleResult/Credentials/SecretAccessKey/text()"s,
+                        session_token: ~x"./AssumeRoleResult/Credentials/SessionToken/text()"s,
+                        expiration: ~x"./AssumeRoleResult/Credentials/Expiration/text()"s,
+                        assumed_role_id: ~x"./AssumeRoleResult/AssumedRoleUser/AssumedRoleId/text()"s,
+                        assumed_role_arn: ~x"./AssumeRoleResult/AssumedRoleUser/Arn/text()"s,
+                        request_id: request_id_xpath())
+
+      {:ok, Map.put(resp, :body, parsed_body)}
+    end
+
+    def parse({:ok, %{body: xml}=resp}, :decode_authorization_message) do
+      parsed_body = xml
+      |> SweetXml.xpath(~x"//DecodeAuthorizationMessageResponse",
+                        decoded_message: ~x"./DecodeAuthorizationMessageResult/DecodedMessage/text()"s,
+                        request_id: request_id_xpath())
+      |> Map.update!(:decoded_message, &Poison.decode!/1)
```

Comment:
So the thing here is that we can't hardcode the JSON codec. Not everyone uses Poison. The configured `json_codec` can be found in the config. You'll need to add that to the arguments of the parse function, which is pretty easy because it's available inside the operation.